src/main/java/uk/gov/register/serialization/RSFExecutor.java

```diff
@@ -71,9 +71,14 @@ private RegisterResult validate(RegisterCommand command, Register register, int
         if (commandName.equals("add-item")) {
             validateAddItem(command, rsfLine, hashRefLine);
         } else if (commandName.equals("append-entry")) {
+            // Check that we should still allow system entries
+            if (allowSystemEntries) {
+                allowSystemEntries = register.getTotalStagedEntries(EntryType.user) == 0 ? register.getTotalEntries(EntryType.user) == 0 : false;
```

Comment:
Can we remove the need to call `getTotalStagedEntries(EntryType.user)` here? Could we change the `getTotalEntries(EntryType entryType)` method in `PostgresDataAccessLayer` to read from `stagedEntries` if it needs to?